pinot-core/src/main/java/org/apache/pinot/core/segment/memory/PinotByteBuffer.java

```diff
@@ -326,9 +329,15 @@ public void flush() {
   }
 
   @Override
-  protected void release() {
-    if (((DirectBuffer) _buffer).cleaner() != null) {
-      ((DirectBuffer) _buffer).cleaner().clean();
+  protected void release()
+      throws IOException {
+    if (CleanerUtil.UNMAP_SUPPORTED) {
```

Comment:
Maybe better to log an warning with the reason when CleanerUtil.UNMAP_SUPPORTED is false

Reply:
It's logged in CleanerUtil